### PR TITLE
Fix error

### PR DESCRIPTION
**Pull request by spartucus**
## Description

Fix error, while the resource of `std::shared_ptr` exists, the `expired()` method of `std::weak_ptr` returns `false`, otherwise returns `true`.

## Change List

- Fix error of  `expired()` return value

## Reference

https://en.cppreference.com/w/cpp/memory/weak_ptr/expired

## 说明

修复错误，`std::shared_ptr` 资源尚存在时，`std::weak_ptr` 的 `expired()` 方法返回 `false`, 否则返回 `true`。

## 变化箱单

- 修复了关于 `std::weak_ptr` 的 `expired()` 方法返回值错误

## 参考文献

https://en.cppreference.com/w/cpp/memory/weak_ptr/expired

### CODE REVIEW

File: book/zh-cn/05-pointers.md [hunk 1]
```diff
@@ -175,7 +175,7 @@ int main() {
 
 在上图中，最后一步只剩下 B，而 B 并没有任何智能指针引用它，因此这块内存资源也会被释放。
 
-`std::weak_ptr` 没有 `*` 运算符和 `->` 运算符，所以不能够对资源进行操作，它的唯一作用就是用于检查 `std::shared_ptr` 是否存在，其 `expired()` 方法能在资源未被释放时，会返回 `true`，否则返回 `false`。
+`std::weak_ptr` 没有 `*` 运算符和 `->` 运算符，所以不能够对资源进行操作，它的唯一作用就是用于检查 `std::shared_ptr` 是否存在，其 `expired()` 方法能在资源未被释放时，会返回 `false`，否则返回 `true`。
```
Comment (changkun) [inline book/zh-cn/05-pointers.md:178]: Could you please maybe also fix the error in `book/en-us/05-pointers.md`?

Comment (spartucus) [reply book/zh-cn/05-pointers.md:178]: Sure thing!

Comment (spartucus) [reply book/zh-cn/05-pointers.md:178]: Done!

**Review by changkun — APPROVED** [review]
Thank you so much for your contribution!

**Pull request merged**